src/arraytraits.rs

```diff
@@ -79,6 +81,88 @@ impl<S, S2, D> PartialEq<ArrayBase<S2, D>> for ArrayBase<S, D>
     }
 }
 
+impl<S, S2, D> PartialOrd<ArrayBase<S2, D>> for ArrayBase<S, D>
+    where D: Dimension,
+          S: Data,
+          S2: Data<Elem = S::Elem>,
+          S::Elem: PartialOrd,
+{
+    /// Return `None` if the array shapes `self` and `rhs` are not equal.
+    /// Return appropriate `Option<Ordering>` otherwise.
+    fn partial_cmp(&self, rhs: &ArrayBase<S2, D>) -> Option<Ordering>
+    {
+        // Balk on arguments with different shapes. Should probably
+        // broadcast at some point
+        if self.shape() != rhs.shape() {
+            return None;
+        }
+        // Use partial_cmp from the slice version if contiguous
+        if let Some(self_s) = self.as_slice() {
+            if let Some(rhs_s) = rhs.as_slice() {
+                return self_s.partial_cmp(rhs_s);
+            }
+        }
+
+        let mut cmps = self.iter().zip(rhs.iter()).map(|(a, b)| a.partial_cmp(b));
+        if let Some(first_cmp) = cmps.next() {
+            if cmps.all(|x| x == first_cmp) {return first_cmp}
+        }
+        None
```

Comment:
Sorry I didn't review this before. To make a lexicographical ordering, continue while each pair compares equal. If an element pair is non-equal, that's the result of the whole comparison. If you reach the end, result is Equal (since we already know the length is equal).